### PR TITLE
Python: Update change note for 1.23.

### PR DESCRIPTION
**Pull request by tausbn**
@felicitymay

### CODE REVIEW

File: change-notes/1.23/analysis-python.md [hunk 1]
```diff
@@ -3,7 +3,19 @@
...
 
+Python 3.8 syntax is now supported. In particular, the following constructs are parsed correctly:
+
+- Assignment expressions using the "walrus" operator, such as `while chunk := file.read(1024): ...`.
+- The positional argument separator `/`, such as in `def foo(a, /, b, *, c): ...`.
+- Self-documenting expressions in f-strings, such as `f"{var=}"`.
+
+### General query improvements
+
+In an effort to deprecate the `Object` API (e.g. `ClassObject`) in favour of the
```
Comment (felicitymay) [inline change-notes/1.23/analysis-python.md]: My suggestion here and in the line below is intended to state this more positively.
```suggestion
Following the replacement of the `Object` API (for example, `ClassObject`) in favor of the
```

File: change-notes/1.23/analysis-python.md [hunk 2]
```diff
@@ -3,7 +3,19 @@
...
+`Value` API (e.g. `ClassValue`), many of the standard queries have been updated
```
Comment (felicitymay) [inline change-notes/1.23/analysis-python.md]:
```suggestion
`Value` API (for example, `ClassValue`) in the 1.21 release, many of the standard queries have been updated
```

File: change-notes/1.23/analysis-python.md [hunk 3]
```diff
@@ -20,8 +32,21 @@
...
 | Unreachable code | Fewer false positives | Analysis now accounts for uses of `contextlib.suppress` to suppress exceptions. |
 | `__iter__` method returns a non-iterator | Better alert message | Alert now highlights which class is expected to be an iterator. |
-
+| Explicit return in __init__ method | Fewer false positives | Instances where the `__init__` method returns the value of a call to a procedure are no longer flagged. |
+| Non-iterable used in for loop | Fewer false positives | `__aiter__` is now recognized as an iterator method. |
+| Unused import | Fewer false positives | Instances where a module is used in a forward-referenced type annotation, or only during type checking are no longer flagged. |
+| Module-level cyclic import | Fewer false positives | Instances where one of the links in an import cycle is never actually executed are no longer flagged. |
+| Undefined export | Fewer false positives | Instances where an exported value may be defined in a module that lacks points-to information are no longer flagged. |
+| Unreachable code | Fewer false positives | Unreachable `else` branches that do nothing but `assert` their non-reachability are no longer flagged. |
```
Comment (felicitymay) [inline change-notes/1.23/analysis-python.md]: Please can we have these sorted alphabetically and can you add the query identifiers to the first column.

File: change-notes/1.23/analysis-python.md [hunk 4]
```diff
@@ -20,8 +32,21 @@
 |----------------------------|------------------------|------------|
 | Unreachable code | Fewer false positives | Analysis now accounts for uses of `contextlib.suppress` to suppress exceptions. |
 | `__iter__` method returns a non-iterator | Better alert message | Alert now highlights which class is expected to be an iterator. |
-
+| Explicit return in __init__ method | Fewer false positives | Instances where the `__init__` method returns the value of a call to a procedure are no longer flagged. |
```
Comment (felicitymay) [inline change-notes/1.23/analysis-python.md]: Please make the following change to this column.
```suggestion
| Explicit return in __init__ method | Fewer false positive results | Instances where the `__init__` method returns the value of a call to a procedure are no longer flagged. |
```

File: change-notes/1.23/analysis-python.md [hunk 5]
```diff
@@ -20,8 +32,21 @@
...
+| Unused import | Fewer false positives | Instances where a module is used in a forward-referenced type annotation, or only during type checking are no longer flagged. |
+| Module-level cyclic import | Fewer false positives | Instances where one of the links in an import cycle is never actually executed are no longer flagged. |
+| Undefined export | Fewer false positives | Instances where an exported value may be defined in a module that lacks points-to information are no longer flagged. |
+| Unreachable code | Fewer false positives | Unreachable `else` branches that do nothing but `assert` their non-reachability are no longer flagged. |
 
 ## Changes to QL libraries
 
 * Django library now recognizes positional arguments from a `django.conf.urls.url` regex (Django version 1.x)
+* Instances of the `Value` class now support the `isAbsent` method, indicating
+  whether the `Value` in question is missing points-to information, but has been
```
Comment (felicitymay) [inline change-notes/1.23/analysis-python.md]: Possibly simpler as:

```suggestion
  whether that `Value` lacks points-to information, but inference
```

File: change-notes/1.23/analysis-python.md [hunk 6]
```diff
@@ -20,8 +32,21 @@
...
+  inferred to likely exist anyway. For instance, if a file contains `import
```
Comment (felicitymay) [inline change-notes/1.23/analysis-python.md]:
```suggestion
  suggests that it exists. For instance, if a file contains `import
```

**Comment by tausbn** [comment]
Thank you for the suggestions! 
I will fix the `positives -> positive results`, alphabetisation, and query IDs in a separate commit.

File: change-notes/1.23/analysis-python.md (hunk 3 above)
Comment (tausbn) [reply change-notes/1.23/analysis-python.md]: Done.

File: change-notes/1.23/analysis-python.md (hunk 4 above)
Comment (tausbn) [reply change-notes/1.23/analysis-python.md]: Done.

**Comment by tausbn** [comment]
Sorry about the many force pushes. It seems my local config was a bit messed up (hence the commits from `taus-semmle`). Hopefully it should be resolved now.

**Pull request merged**